python/cuml/fil/fil.pyx

```diff
@@ -228,17 +229,16 @@ cdef class ForestInference_impl():
         cdef cumlHandle* handle_ =\
             <cumlHandle*><size_t>self.handle.getHandle()
 
+        preds = cudf.Series(zeros(n_rows, dtype=np.float32))
```

Comment:
Is there a particular reason to use `cudf.Series` here rather than `rmm.device_array`?

Reply:
I required it to return a cudf series for post processing and as I could not recall a specific reason for using rmm over cudf here originally, I update the code to use cudf.Series here instead of converting the rmm array to cudf series. Please let me know if i missed something and we do require or want it to be rmm.

Reply:
This part will get updated soon to use Array when the shim PR gets merged

Reply:
@canonizer the PR that salon I mentions is #1635 that will homogenize all our python data structures and give automatic output format conversion. It is scheduled that all or most algos will be using it by 0.13 release. So for the meantime this might be fine

Reply:
This pull request is not directly related to the FIL Python wrapper, hence my comment.

Also, this makes the `preds != None` option not working, so this needs to be addressed as well.

@Salonijain27 If you need to return a `cudf.Series`, wouldn't calling `fil_model.predict(data, preds=cudf.Series(...))` be better?